pkg/operator/ceph/object/controller.go

```diff
@@ -257,12 +230,18 @@ func (r *ReconcileCephObjectStore) reconcile(request reconcile.Request) (reconci
 		}
 		objCtx, err := NewMultisiteContext(r.context, r.clusterInfo, cephObjectStore)
 		if err != nil {
-			return reconcile.Result{}, *cephObjectStore, errors.Wrapf(err, "failed to check for object buckets. failed to get object context")
+			return reconcile.Result{}, *cephObjectStore, errors.Wrapf(err, "failed to get object context")
 		}
 		opsCtx, err := NewMultisiteAdminOpsContext(objCtx, &cephObjectStore.Spec)
 		if err != nil {
-			return reconcile.Result{}, *cephObjectStore, errors.Wrapf(err, "failed to check for object buckets. failed to get admin ops API context")
+			return reconcile.Result{}, *cephObjectStore, errors.Wrapf(err, "failed to get admin ops API context")
 		}
+
+		// ignore any errors coming from this for the deletion case since we could get into a
+		// partially-deleted case where RGWs aren't responding. Deletion will be blocked later if
+		// the bucket fails to be deleted and the RGWs are running.
+		_ = removeDeprecatedHealthCheckBucket(r.clusterInfo.Context, opsCtx, cephObjectStore)
```

Comment:
Based on the separate thread, don't we need to fail and retry the reconcile here instead of ignoring the error?

Reply:
But I see the comment mentions why the partially deleted state would be less desirable.